torch/cuda/comm.py

```diff
@@ -42,7 +42,8 @@ def broadcast_coalesced(tensors, devices, buffer_size=10485760):
 def reduce_add(inputs, destination=None):
     """Sums tensors from multiple GPUs.
 
-    All inputs should have matching shapes.
+    All inputs should have matching shapes, dtype, and layout. The output tensor
```

Comment:
Do we want to mention they must be contiguous and dense?

Reply:
actually they don't have to be contiguous and dense. if they are not, this will go to the non-nccl codepath

Reply:
ah, I see